Add i18n

Lisiadito opened the pull request:
Fixes #20 

I have a german translation in progress as well but I wanted to get this reviewed before putting to much effort in the german translation.

lorcalhost commented:
Hi @Lisiadito , I see that you chose to include a file `./config/lang.cfg` for language selection.  
Changing values inside this file would interfere with the update-process (the update button runs `git pull`).  
I think a better solution should be found for language selection:
- CLI flags
- Manually created file / created on first startup
- Langauge selection through bot menus
- Other ways (?)

Other than that thanks for the PR, I'll check it out more throughly when I have time

Lisiadito commented:
Ah I didn't think about that. Makes sense. Yeah will think about it. Do you have a preference? 
Tomorrow I will start using that branch and test if I actually missed something 😁

lorcalhost commented:
No preference, each option I listed has its pros and cons.  
Feel free to suggest alternatives.

Lisiadito commented:
@lorcalhost I still find some mistakes I made and I'll fix them the next days. I let you know when the branch is ready for a review. And I can squash all the bugfix commits when I'm done.

Lisiadito commented:
@lorcalhost Ok I fixed everything I've found. I use this branch now with my bot so I'll see if anything more shows up but I would say it's ready for review.

Lisiadito commented:
Obviously I still have to decide for a solution on how to set the language.

lorcalhost commented:
Thank you for your hard work!

shhhmel commented on i18n/en.yml:
```diff
@@ -0,0 +1,153 @@
+stop_bot_before_edit_coin_list: '⚠ Please stop Binance Trade Bot before editing the coin list\.'
+coin_list_is: "Current coin list is:"
+reply_coin_list: '_*Please reply with a message containing the updated coin list*_\.'
+coin_list_error: '❌ Unable to find user configuration file at `%{path}`\.'
```
`user configuration file` -> `coin list`

Lisiadito replied:
fixed

shhhmel commented on i18n/en.yml:
```diff
@@ -0,0 +1,153 @@
+deleted_db: '✔ Successfully deleted database file\.'
+delete_db_error: '❌ Unable to delete database file\.'
+clear_log_error: '❌ Unable to clear log file\.'
+exit_no_change: '👌 Exited without changes\.'
```
Duplicate of line 128

Lisiadito replied:
fixed

Lisiadito commented:
@lorcalhost soooo finally managed to fix the way the language is selected. Decided for the command line flag. Sorry for the crazy delay I was super busy.

@shhhmel thanks to you for reviewing and finding some issues. You can merge this branch now again into yours. You need to update some stuff:
Readme, options for the commandline flag help text and I removed one and added one keyword to the en.yml file.
But you will see.

lorcalhost commented:
@Lisiadito thank you for working on this even after all this time.
I'll check out the update

Lisiadito commented:
Oh I just saw that I missed the next_coin funtion. I came in with the merge of the main branch. I add i18n to that and push it in a few minutes

Lisiadito commented:
I fixed the order of the next_coin results. was just one line and it was bothering me a lot.

titulebolide commented:
@Lisiadito is this PR ready to be merged according to you? If so I will start to check it :)

titulebolide commented:
@Lisiadito up :) i would like to have your advice on it before merging

Lisiadito commented on btb_manager_telegram/buttons.py:
```diff
@@ -336,12 +350,13 @@ def next_coin():
                 query = cur.fetchall()
 
                 m_list = []
+                query = sorted(query, key=itemgetter(3), reverse=True)
```
Hey sorry I didn't get a notification that you commented. I don't think there was a specific reason for using that way other than that I'm not fluent in python. I can change it and test if it behaves the same.

titulebolide replied:
Hey thanks for your reply @Lisiadito :)
Using lambda function in this case makes things clearer to me, as itemgetter is not common to be seen in python
Moreover it removes an aditionnal import

Lisiadito replied:
Changed it.

titulebolide commented:
@Lisiadito did you had the change tested in any case?

titulebolide commented:
Great work. Thanks for your contribution @Lisiadito !

titulebolide commented:
@all-contributors please add @Lisiadito for code, idea, translation, doc

titulebolide commented:
@all-contributors please add @Lisiadito for code, idea, translation, doc
